test/ComponentInstallerTest.php

```diff
@@ -571,4 +571,65 @@ public function testOnPostPackageUninstallCanRemovePackageArraysFromConfiguratio
         $this->assertNotContains('Some\Component', $config);
         $this->assertNotContains('Other\Component', $config);
     }
+
+    public function testModuleIsAppended()
+    {
+        $this->createApplicationConfig(
+            '<' . "?php\nreturn [\n    'modules' => [\n        'Some\Component',\n    ]\n];"
+        );
+
+        $package = $this->prophesize(PackageInterface::class);
+        $package->getName()->willReturn('some/module');
+        $package->getExtra()->willReturn(['zf' => [
+            'module' => 'Some\\Module',
+        ]]);
+
+        $operation = $this->prophesize(InstallOperation::class);
+        $operation->getPackage()->willReturn($package->reveal());
+
+        $event = $this->prophesize(PackageEvent::class);
+        $event->isDevMode()->willReturn(true);
+        $event->getOperation()->willReturn($operation->reveal());
+
+        $this->io->ask(Argument::that(function ($argument) {
+            if (! is_array($argument)) {
+                return false;
+            }
+
+            if (! strstr($argument[0], "Please select which config file you wish to inject 'Some\Module' into")) {
+                return false;
+            }
+
+            if (! strstr($argument[1], 'Do not inject')) {
+                return false;
+            }
+
+            if (! strstr($argument[2], 'application.config.php')) {
+                return false;
+            }
+
+            return true;
+        }), 0)->willReturn(1);
```

Comment:
Yes, but then there is something else wrong. Because I took this part from another testcase and the `$modules` variable still holds two entries.